### ISSUE TITLE
[CTM] Property weight does not work

### ISSUE THREAD

**Issue by SQwatermark** [sp614x/optifine#4294]
## Description of Issue
 Property weight does not work. Only the path of the properties file can determing whith one to use. 

But the document says: 

**ctm.properties:**
(Optional) If multiple properties files match the same block, the highest
weighted one is used.  In the event of a tie, the properties filenames are
compared next.  The default weight is 0.
```properties
weight=<integer>
```

I read ConnectedProperties.class but infact it doesn't parse the property "weight".

## OptiFine Version
preview_OptiFine_1.15.2_HD_U_G1_pre30

**Comment by sp614x** [comment]
This is correct, the `weight` property is not checked.

**Comment by Cloud7050** [comment]
On 1.15-pre30, it appears that properties filename comparison is not working, but non-negative integer weights still work. I have been trying to make my pack work with another pack which does not specify weights. It defaults to 0, so to work around that I have tried negative weights and prepending my properties file with last-sorted characters to my pack, but only adding weight=1 to said other pack is working.

**Comment by BrokenDoop** [comment]
Yeah, weight doesn't do anything for me either, trying to get my Crimson Nylium overlay texture to have higher priority than my Sand overlay texture, it only cares about the file names which I don't like. (all my textures and respective properties files have their own sub folder so I could add a letter to the start of the folder name to change its priority but I really don't want to do that.)
